egi_notebooks_hub/d4science.py

```diff
@@ -449,9 +483,53 @@ def profile_list(self, spawner):
         self.log.debug("Profiles: %s", sorted_profiles)
         return sorted_profiles
 
-    async def pre_spawn_hook(self, spawner):
-        # add volumes as defined in the D4Science info sys
+    def _configure_ophidia(self, spawner):
+        if not self.use_ophidia:
+            return
+        chosen_profile = spawner.user_options.get("profile", "")
+        if "ophidia" in chosen_profile:
+            ophidia_mounts = [
+                m for m in spawner.volume_mounts if m["name"] != "workspace"
+            ]
+            ophidia = {
+                "name": "ophidia",
+                "image": self.ophidia_image,
+                "volumeMounts": ophidia_mounts,
+            }
+            spawner.extra_containers.append(ophidia)
```

Comment:
shouldn't this go after setting the  environment ? ( not sure)

Reply:
you mean the lines below? should be the same as the container will be created in another function, here we are just preparing things

Reply:
ok !